pkg/cloud/vsphere/actuators/cluster/actuator.go

```diff
@@ -241,10 +269,10 @@ func (a *Actuator) reconcileReadyState(ctx *context.ClusterContext) error {
 }
 
 func (a *Actuator) deleteKubeConfig(ctx *context.ClusterContext) error {
-	secretName := remotev1.KubeConfigSecretName(ctx.Cluster.Name)
-	if err := a.coreClient.Secrets(ctx.Cluster.Namespace).Delete(secretName, &metav1.DeleteOptions{}); err != nil {
-		if !apierrors.IsNotFound(err) {
-			return errors.Wrapf(err, "error deleting kubeconfig secret for %q", ctx)
+	if err := a.coreClient.Secrets(ctx.Cluster.Namespace).Delete(remotev1.KubeConfigSecretName(ctx.Cluster.Name), &metav1.DeleteOptions{}); err != nil {
+		if !apierrors.IsGone(err) && !apierrors.IsNotFound(err) {
+			ctx.Logger.Error(err, "error deleting kubeconfig secret for target cluster")
+			return &clusterErr.RequeueAfterError{RequeueAfter: constants.DefaultRequeue}
```

Comment:
return the requeue error from actuator `Delete`?

Reply:
Good call. I just checked, neither the `Cluster.Delete` or `Machine.Delete` check for the requeue error.